src/types.ts

```diff
@@ -64,10 +66,17 @@ export interface TxnContext {
     aborted?: boolean | null;
     keys?: string[] | null;
     preds?: string[] | null;
+    readOnly: boolean;
```

Comment:
Is this for the server's response? I don't think we're ever reading these fields, are you sure you need RO/BE flags *both* here and in TxnOptions?